Fix nightly E2E tests & several small improvements to pagination handling

UnseenWizzard opened the pull request:
#### What this PR does / Why we need it:
- fix a small mistake in the pagination E2E test assertion (804aff68f991fc43f7492ec79128ee61c05ce74b)
- fix when a warning is logged about mismatch between received paginated entries and expected total count

#### Special notes for your reviewer:
Review per commit. The actual fix is in the second commit, the rest are small improvements found along the way,

#### Does this PR introduce a user-facing change?
YES - no incorrect warning is logged if Settings filtering results in less objects being returned from the client's List operation.

warber commented on pkg/rest/list.go:
```diff
@@ -28,8 +28,16 @@ import (
+// AddEntriesToResult is a function which should parse an API response body and append the returned entries to a result slice
+// Handling the parsing, any possible filtering and owning and filling the result list is left to the caller of ListPaginated,
+// as it might differ notably between client implementations.
+// The function MUST return the number of entires it has parsed from the received API payload body. This is used to validate
```
```suggestion
// The function MUST return the number of entries it has parsed from the received API payload body. This is used to validate
```

UnseenWizzard replied:
Thanks, correct on the commit that introduced it.

warber reviewed (dismissed):
lgtm

UnseenWizzard commented:
@warber I've addressed the typo (thx) via fixup and rebased on the changes from you Platform/Classic URL improvement. 
Please take another look/hit approve again :)
